lib/extractor/index.ts

```diff
@@ -75,19 +75,34 @@ function layersWithLatestFileModifications(
   layers: ExtractedLayers[],
 ): ExtractedLayers {
   const extractedLayers: ExtractedLayers = {};
+  const filesToRemove: Set<string> = new Set();
+
   // TODO: This removes the information about the layer name, maybe we would need it in the future?
   for (const layer of layers) {
     // go over extracted files products found in this layer
     for (const filename of Object.keys(layer)) {
-      // file was not found
-      if (!Reflect.has(extractedLayers, filename)) {
+      if (isDeletedFile(filename)) {
+        filesToRemove.add(filename.replace(/.wh./, ""));
+      }
+
+      if (filesToRemove.has(filename)) {
```

Comment:
It is not quite clear from looking at the code that you are working with both the deleted and not deleted file, it would be nice to either extract a method or add some comment :-)

Reply:
I have tried to add some comments,
since it's located between loops adding another function isn't ideal for the returned value